core/src/main/java/org/sonarsource/sonarlint/core/plugin/PluginManifest.java

```diff
@@ -0,0 +1,388 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.plugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.builder.ReflectionToStringBuilder;
+
+/**
+ * This class loads Sonar plugin metadata from JAR manifest.
+ */
+public final class PluginManifest {
+
+  public static final String KEY = "Plugin-Key";
+  public static final String MAIN_CLASS = "Plugin-Class";
+  public static final String NAME = "Plugin-Name";
+  public static final String DESCRIPTION = "Plugin-Description";
+  public static final String ORGANIZATION = "Plugin-Organization";
+  public static final String ORGANIZATION_URL = "Plugin-OrganizationUrl";
+  public static final String LICENSE = "Plugin-License";
+  public static final String VERSION = "Plugin-Version";
+  public static final String SONAR_VERSION = "Sonar-Version";
+  public static final String DEPENDENCIES = "Plugin-Dependencies";
+  public static final String HOMEPAGE = "Plugin-Homepage";
+  public static final String TERMS_CONDITIONS_URL = "Plugin-TermsConditionsUrl";
+  public static final String BUILD_DATE = "Plugin-BuildDate";
+  public static final String ISSUE_TRACKER_URL = "Plugin-IssueTrackerUrl";
+  public static final String REQUIRE_PLUGINS = "Plugin-RequirePlugins";
+
+  /**
+   * @since 0.3
+   */
+  public static final String USE_CHILD_FIRST_CLASSLOADER = "Plugin-ChildFirstClassLoader";
+
+  /**
+   * @since 1.1
+   */
+  public static final String BASE_PLUGIN = "Plugin-Base";
+
+  /**
+   * @since 1.3
+   */
+  public static final String IMPLEMENTATION_BUILD = "Implementation-Build";
+
+  /**
+   * @since 1.4
+   */
+  public static final String SOURCES_URL = "Plugin-SourcesUrl";
+
+  /**
+   * @since 1.4
+   */
+  public static final String DEVELOPERS = "Plugin-Developers";
+
+  private String key;
+  private String name;
+  private String mainClass;
+  private String description;
+  private String organization;
+  private String organizationUrl;
+  private String license;
+  private String version;
+  private String sonarVersion;
+  private String[] dependencies;
+  private String homepage;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename field "homepage" to prevent any misunderstanding/clash with field "HOMEPAGE" defined on line 46. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1845)